cEP-0018.md

```diff
@@ -22,6 +22,38 @@ they can write advanced linting bears. This will aim at supporting the more
 flexible visitor based method of AST traversal (as opposed to listener based
 mechanisms).
 
+ANTLR has a `HiddenTokenStream` which recieves all the tokens irrelevant to the
+grammar such as whitespaces. These are not passed on to the parser and as a
+result are not returned when simply fetching text from a node in parse tree.
+
+Consider the code:
+
+```python
+var_x = 5
+if var_x:
+    print('Variable x is', var_x)
+else:
+    print('Variable x is not set')
+```
+
+After converting this to parsetree and fetching source back using
+[this](https://gist.github.com/virresh/643cd1a2d9149668980814d970f98fbc#file-roundtripper-py-L13)
+code we get the following output:
+
+```python
+var_x=5
+ifvar_x:
+print('Variable x is',var_x)
+else:
+print('Variable x is not set')
+```
+
+This prevents us from easily fetching source code.
+
+Since ANTLR has poor support for retrieving original source code from the
+parse tree, the feature to fix code is out of scope for this project, and will
```

Comment:
A "fix" does not require doing a 100% round-trip.  It requires the ability to modify a small chunk of a source file.

 Does ANTLR provide line and column offsets of the nodes of the parse tree?  If so, fixes are *possible*.

coala already has a loaded copy of the source file.  If ANTLR provides line and column number, many parts of the above scenario are able to be modified without any problems.

Even the string, changing that string in your example only requires knowing basic string quoting rules.

Where I believe things get messy is modifying any node which crosses the line barrier, especially multiline strings, because ANTLR will only provide start (and end) offsets, and applying a fix will fail in those instances because the linter believes the string looks like "a b c" but in reality the file contains "a\\\nb\\\nc" and a diff changing it to "a d c" wont apply cleanly.

The same problem *might* even exist in the simple example given above, as the `if...else..` spans multiple lines.  Does the `else` have its own line/col offsets in the parse tree?  If not, it is a multi-line construct where ANTLR has made it difficult to push changes back into the source tree.

If round-tripping isnt possible, the ability to create a fix after a problem has been detected depends on whether the fixer still needs to re-parse the lines to apply the fix.  If so, too much context was lost.  For a fixer which modifies strings, it is OK that the fixer needs to know the rules of a string *inside* the Python string.  However a fixer of strings shouldnt need to reparse the lines and know Python line continuation `\` rules (even if they are simple) because that is beyond the scope of the fixer.

So you need to adjust your example to show why this is actually hard.

And your example shouldnt only show that strings are hard.
If only strings are hard, you can only legitimately say your project shouldnt fix strings.

You need to *prove* your project couldnt feasibly fix *any* code :P 
(or your project needs to include a fixer for a scenario that you can correctly fix, even if it is very basic)